libpod/runtime_ctr.go

```diff
@@ -60,7 +60,7 @@ func (r *Runtime) NewContainer(spec *spec.Spec, options ...CtrCreateOption) (ctr
 		}
 	}
 	defer func() {
-		if err != nil {
+		if err != nil && ctr.pod != nil {
```

Comment:
Should this be c.pod?

Reply:
c is probably going to be nil when this is called, because we always return nil, err when an error occurs. If I could, I wouldn't have a variable assigned to the container we return at all, only the error, but I think Go won't let me do that.